### ISSUE TITLE
Python 3.13 new `__firstlineno__` class attribute

### ISSUE THREAD

**Issue by koutoftimer** [microsoft/pyright#9484]
https://github.com/python/cpython/issues/118465 introduces new class attribute `__firstlineno__`, which pyright doesn't support.

An example code work fine in python3.13 runtime 

```python
class A: ...


print(f"{A.__firstlineno__=}")
```

but pyright shows an error

```
:!pyright run.py                                                                                         
                                                                                                         
/tmp/tmp.A9F2mvqmhp/run.py                                                                               
  /tmp/tmp.A9F2mvqmhp/run.py:4:12 - error: Cannot access attribute "__firstlineno__" for class "type[A]" 
    Attribute "__firstlineno__" is unknown (reportAttributeAccessIssue)                                  
1 error, 0 warnings, 0 informations
```

I've been using `pyright 1.1.389` via neovim/mason/pyright extension.

```
:!pyright --version                                                                                                                                                                                                
                                                                                                                                                                                                                   
pyright 1.1.389
```

**Comment by erictraut** [comment]
Pyright doesn't have hard-coded knowledge of symbols like this. If you'd like to see this supported, it would need to be added to the [typeshed stdlib stubs](https://github.com/python/typeshed), which are used by all Python type checkers including pyright. Feel free to submit a feature request or PR to typeshed.
